plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -263,27 +265,19 @@ public void testWriteLargeFiles()
                         "    regionkey bigint) " +
                         "    WITH (partitioned_by=ARRAY['regionkey'])",
                 testTable));
-        String largeRowExpression = "array_join(transform(sequence(1, 70), x-> array_join(repeat(comment, 1000), '')), '')";
-        computeActual(format("INSERT INTO " + testTable + " SELECT %s, %s, regionkey FROM tpch.tiny.nation WHERE regionkey = 1", largeRowExpression, largeRowExpression));
+
+        long partSizeInBytes = HIVE_S3_STREAMING_PART_SIZE.toBytes();
 
         // Exercise different code paths of Hive S3 streaming upload, with upload part size 5MB:
         // 1. fileSize <= 5MB (direct upload)
+        assertFileSizeRange(testTable, 50, 0, partSizeInBytes);
+
         // 2. 5MB < fileSize <= 10MB (upload in two parts)
+        assertFileSizeRange(testTable, 100, partSizeInBytes + 1, partSizeInBytes * 2);
+
         // 3. fileSize > 10MB (upload in three or more parts)
-        query(format("SELECT DISTINCT (\"$file_size\" - 1) / (5 * 1024 * 1024) FROM %s ORDER BY 1", testTable))
-                .assertThat()
-                .skippingTypesCheck()
-                .containsAll(resultBuilder(getSession())
-                        .row(0L)
-                        .row(1L)
-                        .row(2L)
-                        .build());
-        query(format("SELECT SUM(LENGTH(col1)) FROM %s", testTable))
-                .assertThat()
-                .skippingTypesCheck()
-                .containsAll(resultBuilder(getSession())
-                        .row(500L * 70 * 1000)
-                        .build());
+        assertFileSizeRange(testTable, 150, partSizeInBytes * 2 + 1, partSizeInBytes * 3);
```

Comment:
How are we sure that we will not get only small files if we end up having many writer tasks?
I am probably missing sth.

Reply:
My understanding even if we have multiple writer tasks, the distribution is still on a row-basis. Given that we write only one row, then there will be only one file.

Reply:
Oh yeah - sure :) Makes total sense.